### ISSUE TITLE
fromPromise unset() method

### ISSUE THREAD

**Issue by fabien** [dwstech/muster#28]
**Is your feature request related to a problem? Please describe.**

There's currently no `unset` node to perform a delete/remove through `fromPromise` like this:

```javascript
const app = muster({
    items: {
        [match(types.string, 'id')]: fromPromise({
            get: function(id) { // Fetch item
                ...
            },
            set: function(id, data) { // Update item
                ...
            },
            unset: function(id) { // Remove item
                ...
            }
        }),
    }
});

app.resolve(unset(ref('items', 'my-id'))).then(function() {
    console.log('Deleted item');
});

```

**Comment by dws-tomasz** [comment]
Hi @fabien, how about renaming `unset` to `reset`, which is an existing Muster operation. In that case the code would look like this:
```javascript
const app = muster({
    items: {
        [match(types.string, 'id')]: fromPromise({
            get: function({ id }) { // Fetch item
                ...
            },
            set: function({ id, data }) { // Update item
                ...
            },
            reset: function({ id }) { // Remove item
                ...
            }
        }),
    }
});

app.resolve(reset(ref('items', 'my-id'))).then(function() {
    console.log('Item deleted');
});
```

**Comment by fabien** [comment]
Hi @dws-tomasz that sounds good as well, although it doesn't carry the same semantics I think. Is one easier to implement than the other? Otherwise, it's worth considering `unset` or even `delete`. Thanks!

**Comment by dws-tomasz** [comment]
You're totally right about the semantics. The only reason why I'd like to avoid adding a new node is that the Muster API is large as is, and I'd like to re-purpose as much of it as possible. There's also a `clear` operation, which should fit this purpose quite well:
```javascript
const app = muster({
    items: {
        [match(types.string, 'id')]: fromPromise({
            get: function({ id }) { // Fetch item
                ...
            },
            set: function({ id, data }) { // Update item
                ...
            },
            clear: function({ id }) { // Remove item
                ...
            }
        }),
    }
});

app.resolve(clear(ref('items', 'my-id'))).then(function() {
    console.log('Item deleted');
});
```

**Comment by fabien** [comment]
Agreed, the Muster API is quite large indeed. I think `clear` will fit the bill nicely!

**Comment by dws-dave** [comment]
Closed via #51
